bridge/launchpad/launchpad_api.go

```diff
@@ -0,0 +1,178 @@
+package launchpad
+
+/*
+ * A wrapper around the Launchpad API. The documentation can be found at:
+ * https://launchpad.net/+apidoc/devel.html
+ *
+ * TODO:
+ * - Retrieve all messages associated to bugs
+ * - Retrieve bug status
+ * - Retrieve activity log
+ * - SearchTasks should yield bugs one by one
+ *
+ * TODO (maybe):
+ * - Authentication (this might help retrieving email adresses)
+ */
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+)
+
+const apiRoot = "https://api.launchpad.net/devel"
+
+// Person describes a person on Launchpad (a bug owner, a message author, ...).
+type LPPerson struct {
+	Name  string `json:"display_name"`
+	Login string `json:"name"`
+}
+
+// Caching all the LPPerson we know.
+// The keys are links to an owner page, such as
+// https://api.launchpad.net/devel/~login
+var personCache = make(map[string]LPPerson)
+
+func (owner *LPPerson) UnmarshalJSON(data []byte) error {
+	type LPPersonX LPPerson // Avoid infinite recursion
+	var ownerLink string
+	if err := json.Unmarshal(data, &ownerLink); err != nil {
+		return err
+	}
+
+	// First, try to gather info about the bug owner using our cache.
+	if cachedPerson, hasKey := personCache[ownerLink]; hasKey {
+		*owner = cachedPerson
+		return nil
+	}
+
+	// If the bug owner is not already known, we have to send a request.
+	req, err := http.NewRequest("GET", ownerLink, nil)
```

Comment:
It's a little odd to have an `UnmarshalJSON` function do a network request, as they are usually just about decoding the data. Not wrong though, no worries.

Reply:
Yeah, I was not sure how to do this. I could store the owner link in the Bug structure, and turn it into a Person structure later, but that is not super convenient.